### PR TITLE
List & Show forwarders command

### LINKED ISSUE #4436: Create clap command to show the details of an existing Forwarder on a node
### Current behavior

Missing command for `forwarder show`.

### Desired behavior

Implement a command that retrieves from a running node the information of a `forwarder` given its name.

```bash
ockam forwarder show [name] --at [node-name]
```

Required args:
- name

Optional args:
- at: defaults to the default node

Output: same as described at [`forwarder create`](https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_command/src/forwarder/create.rs#L80).

### LINKED ISSUE #4441: Create clap command to list the existing forwarders on a node
### Current behavior

Missing command for `forwarder list`.

### Desired behavior

Implement a command that retrieves from a running node the list of registered forwarders.

```bash
ockam forwarder list --at [node-name]
```

Required args: none

Optional args:
- at: defaults to the default node. Use the `NodeOpts` struct to define this argument.

Output: list of items with the same information per item as described at [`forwarder show`](https://github.com/build-trust/ockam/tree/develop/implementations/rust/ockam/ockam_command/src/forwarder/create.rs#L81).

Similar command: [`tcp-inlet list`](https://github.com/build-trust/ockam/blob/fba16afa8ca1a2130819737d6804480a31996a5b/implementations/rust/ockam/ockam_command/src/tcp/inlet/list.rs#L23)

### PR DESCRIPTION
**Pull request by nidnogg**
## Current behavior

Currently, whenever a forwarder is created, there is no way to directly retrieve it. 

## Proposed changes

- Added a `forwarders` registry `BTreeMap<String, ForwarderRegistryInfo>` to `ockam_api/src/nodes/registry.rs` to make forwarders accessible;
- Added `forwarder list` to solve https://github.com/build-trust/ockam/issues/4441;
- Added `forwarder show` to solve https://github.com/build-trust/ockam/issues/4436;
- Added tests to both commands in `unit.bats`

## Checks

- [X] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [X] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [X] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [X] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository. The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/edit/main/CONTRIBUTORS.csv) file in the github web UI and create a PR, this will mark the commit as verified.

### CODE REVIEW

File: implementations/rust/ockam/ockam_api/src/nodes/registry.rs [hunk 1]
```diff
@@ -122,6 +122,39 @@ impl KafkaServiceInfo {
         &self.kind
     }
 }
+
+#[derive(Debug, Clone)]
+pub(crate) struct ForwarderRegistryInfo {
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_api/src/nodes/registry.rs]: What's the benefit of using this struct over `RemoteForwarderInfo`? 

If there isn't I'd remove it and store the `RemoteForwarderInfo` in the registry directly.

File: implementations/rust/ockam/ockam_api/src/nodes/models/forwarder.rs [hunk 2]
```diff
@@ -106,3 +112,33 @@ impl<'a> From<RemoteForwarderInfo> for ForwarderInfo<'a> {
...
+            worker_address: inner.worker_address().to_string().into(),
+        }
+    }
+}
+
+/// Response body when returning a list of Forwarders
+#[derive(Debug, Clone, Decode, Encode)]
+#[rustfmt::skip]
+#[cbor(map)]
+pub struct ForwarderList<'a> {
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_api/src/nodes/models/forwarder.rs]: We have some mixed usages on lists endpoints, but I'm pretty sure you could return just the `Vec<ForwarderInfo<'a>>` as part of the CBOR body. You could then remove this struct.

**Review by adrianbenavides — DISMISSED** [review]
It's looking great @nidnogg!

File: implementations/rust/ockam/ockam_api/src/nodes/models/forwarder.rs (hunk 2 above)
Comment (nidnogg) [reply implementations/rust/ockam/ockam_api/src/nodes/models/forwarder.rs]: I mirrored the `InletList` and `OutletList` structs used for the portals' `list` commands - perhaps we could refactor these out as well? They had a very similar same format. For now I've removed this in the latest commit

File: implementations/rust/ockam/ockam_api/src/nodes/registry.rs (hunk 1 above)
Comment (nidnogg) [reply implementations/rust/ockam/ockam_api/src/nodes/registry.rs]: Fixed in latest commit - it was mostly there to be consistent with the other registry-exclusive structs like `InletInfo` and `OutletInfo`. I wonder if those could be similarly refactored?

**Review by SanjoDeundiak — APPROVED** [review]
Looks great! Merging
cc @adrianbenavides

**Pull request merged**